ddp-workspace/projects/ddp-sdk/src/lib/components/address/addressEmbedded.component.ts

```diff
@@ -186,17 +196,18 @@ export class AddressEmbeddedComponent implements OnDestroy, OnInit {
       showSuggestion: false,
       enteredAddress: null,
       suggestedAddress: null,
+      formWarningMessages: [],
       formErrorMessages: [],
       fieldErrors: []
     };
 
     this.state$ = this.stateUpdates$.pipe(
       startWith(initialState),
       scan((acc: ComponentState, update) => ({ ...acc, ...update })),
+      tap(state =>  console.debug('New embeddedComponentState$=%o', state)),
       shareReplay(1)
     );
-
-    this.state$.subscribe((state) => console.debug('New embeddedComponentState$=' + JSON.stringify(state)));
+    this.state$.subscribe();
```

Comment:
Hm, do we need to take care of unsubscribing from the observable when we use shareReplay?

Also, could you move this.initializeComponentState(); call in the ngOnInit?

Reply:
I had not unsubscribed properly in this spot and another place. Thnx!
930065a72fac5d10a0edf8522c5c00852929b0ac
Regarding the `initializeComponentState()` as currently configured it has to happen in constructor.